pkg/reconciler/knative/resources/knative_configuration.go

```diff
@@ -34,26 +38,26 @@ const (
 )
 
 type ConfigurationBuilder struct {
-	frameworksConfig *v1alpha1.FrameworksConfig
+	frameworksConfig  *v1alpha1.FrameworksConfig
+	credentialBuilder *credentials.CredentialBuilder
 }
 
-func NewConfigurationBuilder(config *v1.ConfigMap) *ConfigurationBuilder {
+func NewConfigurationBuilder(client client.Client, config *v1.ConfigMap) ConfigurationBuilder {
```

Comment:
What's the best practice here for returning *ConfigurationBuilder vs ConfigurationBuilder